### DISCUSSION TITLE
updateMany records by id but with different data

### DISCUSSION THREAD

**Discussion by YassinEldeeb** [prisma/prisma#8514] [Q&A]
So I want to updateMany notification records and I've all of their Ids and the date I wanna update, but updateMany only allows me to provide a certain criteria `where` object and a single `data` object that applies to all of them, but what if I want to update records by Ids but assign different data object for each.
```
[
  {
    id: 'ckrrcq9dr0000dwu68ecl3pft',
    message: 'New Posts from Yassin and Amin',
    notifiedUserId: 'ckrpcao2r0002lou6ccypt2cq',
    type: 'newPosts',
    url: '/timeline'
  },
  {
    id: 'ckrrcq9dr0001dwu60234htit',
    message: 'New Post from Yassin',
    notifiedUserId: 'ckrpcb0zd0020lou6vg0z5di0',
    type: 'newPosts',
    url: '/timeline'
  },
  {
    id: 'ckrrcq9dr0002dwu605figrz0',
    message: 'New Posts from Yassin and 4 others',
    notifiedUserId: 'ckrpcb3rs0029lou641p2ki6x',
    type: 'newPosts',
    url: '/timeline'
  }
]
```
I went with deleteMany by Ids then inserting the data again by createMany for now.

**Comment by ryands17** [comment]
@YassinEldeeb 👋 

This isn't possible currently, but we do have a feature request so it would be great if you could add a 👍 on the request:
https://github.com/prisma/prisma/issues/6862

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
